### DISCUSSION TITLE
Questions on bicycle routing

### DISCUSSION THREAD

**Discussion by DoggoOfSpeed** [organicmaps/organicmaps#5244] [Q&A]
So I've been using organic maps to check for possible tags that cause problems for routing (like, for example, bicycle=dismount on a crossing causes the routing to divert the path to the main road only to rejoin with the sidewalk on the other side of the crossing.

So the thing I've noticed is that paths including stairs are given a much higher weight and are thus avoided. This leads to pedestrian routing being better for bikes than actual bicycle routing as carrying a bike up a flight of stairs is much easier than riding up an incline of the same height.

Pedestrian routing example:
![image](https://github.com/organicmaps/organicmaps/assets/28051112/a476a12f-3a2b-476b-b049-1c8aa1f054f7)
Bicycle routing example:
![image](https://github.com/organicmaps/organicmaps/assets/28051112/c67ea084-7b20-48d7-99ae-53c93629cb09)

As you can see, bicycle routing picks a route that is not only longer, but also follows an incline which is harder to scal than the set of stairs connecting the start and end points.

This also leads bikes to avoid underground passages even when they are much better than crossing a very busy intersection:
![image](https://github.com/organicmaps/organicmaps/assets/28051112/bac8b8a2-bebb-4b14-b047-904ca580431e)
Here's how the path should look:
![image](https://github.com/organicmaps/organicmaps/assets/28051112/406e233b-3f8d-453f-b168-fea829a125a8)

Another example on the same intersection:
![image](https://github.com/organicmaps/organicmaps/assets/28051112/9a62599c-f2f7-48d0-88e2-81ce4f978edb)
![image](https://github.com/organicmaps/organicmaps/assets/28051112/fa5407eb-377e-42a7-8378-9fa50a8f6085)

Here is another example of poor bicycle routing, this time not related to stairs:![image](https://github.com/organicmaps/organicmaps/assets/28051112/6d178576-dfbb-40da-93e5-889d3c478cf0)![image](https://github.com/organicmaps/organicmaps/assets/28051112/e3041419-4a7b-46d0-ab54-1ad1606fd969)

Keep in mind that both the path taken by pedestrian routing and the longer path taken by bicycle routing have the same tags, so I'm not sure why the longer one is being picked.

**Comment by biodranik** [comment]
What about adding an "Allow dismounting" route option that will use stairs and crossings when necessary?
Or even better, in our current terms, "Avoid stairs and dismounting" ?

CC @vng

**Reply by DoggoOfSpeed** [reply]
Yeah, I had a similar idea. Maybe if alternative routes are ever implemented, it would be cool to see multiple routes created by using different weights.

**Comment by vng** [comment]
We have bicycle default speeds for:
- steps = 1 km/h
- dismount (same as for footway and pedestrian) = 2 km/h

In the last example (not steps) the app decided to use a longer path because of the elevation penalty heuristic. I suspect that the elevation is the same for both paths. I still have doubts about the current elevation-penalty algorithm and need more time for experiments.

Thanks for the examples.

**Comment by DoggoOfSpeed** [comment]
Oooh! Just found another quirk. ![image](https://github.com/organicmaps/organicmaps/assets/28051112/cf9ab84d-48b8-4c3f-b1cc-dc312a9dc906)

Here you can see that the chosen path crosses a road twice just because on the other side the sidewalk has been marked as bicycle=designated (also, segregated=no). Maybe that would be fine for a longer distance, but for this example it's certainly a worse path.

**Comment by vng** [comment]
Yes, and other engines agree here:
https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=54.90624%2C23.92986%3B54.90901%2C23.92891#map=17/54.90758/23.92980

Default highway=footway/pedestrian (unless otherwise specified) is not allowed for bicycles in most European countries .. How in Lithuania?

**Reply by DoggoOfSpeed** [reply]
Hmm... Well, at least in Lithuania, whatever goes. Cycling isn't popular enough to have a developed network of bicycle paths, so generally people are allowed to bike on footpaths and wherever else they want with little exceptions.

**Comment by fdr** [comment]
Carrying a bicycle...which not the same thing as a mere dismount...is best left an option. Consider the application of a cargo bicycle, or even just a bicycle carrying any awkward cargo.
